[UIAsyncTextInput] fast/forms/caps-lock-indicator-width.html fails with --use-async-uikit-interactions

whsieh opened the pull request:
#### dd07b27ea919520ca582d087b2bb4d414bfca6d4
<pre>
[UIAsyncTextInput] fast/forms/caps-lock-indicator-width.html fails with --use-async-uikit-interactions
<a href="https://bugs.webkit.org/show_bug.cgi?id=265357">https://bugs.webkit.org/show_bug.cgi?id=265357</a>
<a href="https://rdar.apple.com/118809857">rdar://118809857</a>

Reviewed by Megan Gardner.

Fix this failing layout test when async UIKit interactions are enabled; see below.

* Source/WebCore/platform/ios/WebEvent.mm:
(webEventModifierFlags):

Add missing logic to handle `UIKeyModifierAlphaShift` (i.e. caps lock state) by adding the
`WebEventFlagMaskLeftCapsLockKey` modifier flag.

* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView insertionPointColor]):
(-[WKContentView selectionBarColor]):
(-[WKContentView selectionHighlightColor]):
(-[WKContentView textContentTypeForTesting]):

Replace internal calls into the private `-textInputTraits` method here, to avoid hitting our own
release assert.

Canonical link: <a href="https://commits.webkit.org/271157@main">https://commits.webkit.org/271157@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f5eb0ec2d98947bed0c5b2e830ce072651c60175

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/27491 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/6133 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/28742 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/29720 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/25169 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/27962 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/8052 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/3528 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24960 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/27756 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/4924 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/23605 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/4287 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/4451 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/24602 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/30356 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/25122 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/25027 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/30583 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/4467 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/2606 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/28551 "Found 7 new API test failures: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/document/load-events, /WebKitGTK/TestCookieManager:/webkit/WebKitCookieManager/long-expires, /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/selection/listbox, /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/event-listener, /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/selection/menulist, /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/state-changed, /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/value/basic (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/5934 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6621 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4924 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/4855 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
```diff
@@ -4262,17 +4262,29 @@ - (UIColor *)insertionPointColor
+#if HAVE(UI_ASYNC_TEXT_INTERACTION)
+    if (self.shouldUseAsyncInteractions)
+        return self.extendedTraitsDelegate.insertionPointColor;
+#endif // HAVE(UI_ASYNC_TEXT_INTERACTION)
```
Optional nit for uniformity, either just `#endif` here or `#endif // HAVE(UI_ASYNC_TEXT_INTERACTION)` in the next three changes?

whsieh replied:
Ah, I put a `// HAVE(UI_ASYNC_TEXT_INTERACTION)` here just for clarity, since it's already nested inside another `#if - #elif`, but thought the other `#if`s below read clearly enough. I'll remove the `// HAVE(UI_ASYNC_TEXT_INTERACTION)` here for consistency.

nt1m committed:
Remove OverflowClipEnabled preference
https://bugs.webkit.org/show_bug.cgi?id=265347
rdar://118800232

Reviewed by Darin Adler.

overflow: clip; has shipped for more than a year and has become an essential part of the web platform.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/parser/CSSParserContext.cpp:
(WebCore::add):
* Source/WebCore/css/parser/CSSParserContext.h:

Canonical link: https://commits.webkit.org/271140@main

cdumez committed:
Adopt dynamicDowncast<>() further in css/
https://bugs.webkit.org/show_bug.cgi?id=265359

Reviewed by Darin Adler.

* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::positionOffsetValue):
* Source/WebCore/css/calc/CSSCalcValue.cpp:
(WebCore::createCSSIgnoringZeroLength):
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::isCustomIdentValue):
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeFilterFunction):
(WebCore::CSSPropertyParserHelpers::consumeTransformFunction):
(WebCore::CSSPropertyParserHelpers::isGridTrackFixedSized):
* Source/WebCore/css/parser/CSSPropertyParserWorkerSafe.cpp:
(WebCore::parserMode):
* Source/WebCore/css/process-css-properties.py:
* Source/WebCore/css/query/ContainerQueryFeatures.cpp:
* Source/WebCore/css/query/GenericMediaQueryEvaluator.cpp:
(WebCore::MQ::computeLength):
(WebCore::MQ::evaluateRatioComparison):

Canonical link: https://commits.webkit.org/271141@main

cdumez committed:
Stop using Vector::unsafeAppendWithoutCapacityCheck() in WasmFunctionParser.h
https://bugs.webkit.org/show_bug.cgi?id=265161

Reviewed by Darin Adler.

* Source/JavaScriptCore/wasm/WasmFunctionParser.h:
(JSC::Wasm::FunctionParser<Context>::parseExpression):

Canonical link: https://commits.webkit.org/271142@main

nt1m committed:
Remove LineHeightUnitsEnabled preference
https://bugs.webkit.org/show_bug.cgi?id=265346
rdar://118799769

Reviewed by Darin Adler.

This has been enabled for more than a year now, and there is really no reason to turn this feature off.

Remove it to reduce the uses of DeprecatedGlobalSettings.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/css/parser/CSSParserToken.cpp:
(WebCore::cssPrimitiveValueUnitFromTrie):
* Source/WebCore/page/DeprecatedGlobalSettings.h:
(WebCore::DeprecatedGlobalSettings::setLineHeightUnitsEnabled): Deleted.
(WebCore::DeprecatedGlobalSettings::lineHeightUnitsEnabled): Deleted.
* Source/WebKitLegacy/mac/WebView/WebPreferenceKeysPrivate.h:
* Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
(-[WebPreferences lineHeightUnitsEnabled]):
(-[WebPreferences setLineHeightUnitsEnabled:]):
* Source/WebKitLegacy/mac/WebView/WebPreferencesPrivate.h:

Canonical link: https://commits.webkit.org/271143@main

anttijk committed:
Move renderer invalidity into a separate bit, out from Style::Validity enum
https://bugs.webkit.org/show_bug.cgi?id=265311
rdar://problem/118772735

Reviewed by Alan Baradlay.

Make it possible to mark a render subtree for rebuild without needing to recompute the style.

* Source/WebCore/dom/Element.cpp:
(WebCore::Element::invalidateStyleAndRenderersForSubtree):

Also remove unncesssary sibling invalidation. It is a leftover from old pseudo-class invalidation code.

* Source/WebCore/dom/Node.cpp:
(WebCore::Node::adjustStyleValidity):

Make renderer invalidity a style flag.

(WebCore::Node::invalidateStyle):

Also add an invalidation mode for newly inserted nodes. Those will always need to invalidate the renderer and the ancestor chain.

(WebCore::Node::insertedIntoAncestor):
* Source/WebCore/dom/Node.h:
(WebCore::Node::needsStyleRecalc const):
(WebCore::Node::hasInvalidRenderer const):
(WebCore::Node::setHasValidStyle):
* Source/WebCore/dom/Text.cpp:
(WebCore::Text::updateRendererAfterContentChange):
* Source/WebCore/style/StyleInvalidator.cpp:
(WebCore::Style::Invalidator::invalidateIfNeeded):
* Source/WebCore/style/StyleTreeResolver.cpp:
(WebCore::Style::TreeResolver::createAnimatedElementUpdate):
(WebCore::Style::TreeResolver::resolveComposedTree):
* Source/WebCore/style/StyleValidity.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::styleValidityToToString):

Canonical link: https://commits.webkit.org/271144@main

annevk committed:
Remove empty files from Source/ excluding Source/ThirdParty/
https://bugs.webkit.org/show_bug.cgi?id=265369

Reviewed by Wenson Hsieh.

* Source/JavaScriptCore/shell/PlatformWin.cmake: Removed.
* Source/WebCore/storage/StorageQuotaManager.cpp: Removed.
* Source/WebCore/storage/StorageQuotaManager.h: Removed.
* Source/WebCore/workers/service/server/RegistrationDatabase.cpp: Removed.
* Source/WebCore/workers/service/server/RegistrationDatabase.h: Removed.
* Source/WebCore/workers/service/server/RegistrationStore.cpp: Removed.
* Source/WebCore/workers/service/server/RegistrationStore.h: Removed.

Canonical link: https://commits.webkit.org/271145@main

jjgriego committed:
[JSC] Wasm-GC: Struct members should have their natural alignment
https://bugs.webkit.org/show_bug.cgi?id=265322

Reviewed by Justin Michaud.

For 32-bit concurrent JIT (and also for concurrent GC on 64-bit, AIUI), we need
members of these structs that are refs to be 8-byte aligned (for 64-bit atomic
access and to avoid spanning cache lines respectively); also, for performance
reasons, it's probably also good to have the scalar types aligned, anyways.

* Source/JavaScriptCore/wasm/WasmTypeDefinition.cpp:
(JSC::Wasm::StructType::StructType):
* Source/JavaScriptCore/wasm/WasmTypeDefinition.h:
(JSC::Wasm::typeAlignmentInBytes):

Canonical link: https://commits.webkit.org/271146@main

cdumez committed:
Stop using Vector::unsafeAppendWithoutCapacityCheck() in GetStackTraceFunctor
https://bugs.webkit.org/show_bug.cgi?id=265159

Reviewed by Darin Adler.

* Source/JavaScriptCore/interpreter/Interpreter.cpp:
(JSC::GetStackTraceFunctor::GetStackTraceFunctor):
(JSC::GetStackTraceFunctor::operator() const):
(JSC::GetStackTraceFunctor::frameCountInResults const):
(JSC::Interpreter::getStackTrace):
* Source/JavaScriptCore/runtime/StackFrame.h:

Canonical link: https://commits.webkit.org/271147@main

pvollan committed:
Video freezes in PiP mode
https://bugs.webkit.org/show_bug.cgi?id=265085
rdar://118582721

Reviewed by Andy Estes.

The Info.plist file for the GPU process extension is missing required media playback properties.

* Source/WebKit/Shared/AuxiliaryProcessExtensions/GPUExtension-Info.plist:

Canonical link: https://commits.webkit.org/271148@main

jernoble committed:
[Cocoa] Upstream AV1Utilities
https://bugs.webkit.org/show_bug.cgi?id=265062
rdar://118576496

Reviewed by Eric Carlson.

* LayoutTests/media/av1-codec-parameters-expected.txt: Added.
* LayoutTests/media/av1-codec-parameters.html: Added.
* LayoutTests/media/av1-codec-per-level-constraints-expected.txt: Added.
* LayoutTests/media/av1-codec-per-level-constraints.html: Added.
* LayoutTests/media/av1-codec-string-expected.txt: Added.
* LayoutTests/media/av1-codec-string.html: Added.
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/mediacapabilities/VideoConfiguration.idl:
* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/SourcesCocoaInternalSDK.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/platform/graphics/AV1Utilities.cpp: Added.
(WebCore::parseEnumFromStringView):
(WebCore::parseAV1CodecParameters):
(WebCore::createAV1CodecParametersString):
(WebCore::perLevelConstraints):
(WebCore::validateAV1PerLevelConstraints):
(WebCore::parseAV1DecoderConfigurationRecord):
* Source/WebCore/platform/graphics/AV1Utilities.h: Added.
* Source/WebCore/platform/graphics/cocoa/AV1UtilitiesCocoa.h: Copied from Source/WebCore/Modules/mediacapabilities/VideoConfiguration.idl.
* Source/WebCore/platform/graphics/cocoa/AV1UtilitiesCocoa.mm: Added.
(isConfigurationRecordHDR):
(validateAV1Parameters):
* Source/WebCore/platform/graphics/cocoa/MediaEngineConfigurationFactoryCocoa.cpp:
(WebCore::computeMediaCapabilitiesInfo):
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::parseAV1CodecParameters):
(WebCore::Internals::createAV1CodecParametersString):
(WebCore::Internals::validateAV1PerLevelConstraints):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:

Canonical link: https://commits.webkit.org/271149@main

pvollan committed:
[iOS][GPUP] Allow loading frameworks inside the bundle
https://bugs.webkit.org/show_bug.cgi?id=265131
rdar://118652428

Reviewed by Brent Fulgham.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.GPU.sb.in:

Canonical link: https://commits.webkit.org/271150@main

martadarbinyan committed:
[Gardening]: REGRESSION ( 271130@main ): [ macOS wk1 ] 4 tests in http/tests/security/mixedContent are a consistent failure
https://bugs.webkit.org/show_bug.cgi?id=265388
rdar://118836741

Unreviewed test gardening.

Adding test expectation

* LayoutTests/platform/mac-wk1/TestExpectations:

Canonical link: https://commits.webkit.org/271151@main

pvollan committed:
Change install location of WebKit extension processes
https://bugs.webkit.org/show_bug.cgi?id=265075
rdar://118590395

Reviewed by Brent Fulgham.

* Source/WebKit/Configurations/BaseExtension.xcconfig:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/271152@main

rwlbuis committed:
[LBSE] Add RenderSVGResourceMasker
https://bugs.webkit.org/show_bug.cgi?id=264049

Reviewed by Nikolas Zimmermann.

Implement the second SVG resource in LBSE: masking (RenderSVGResourceMasker).
Mask content invalidations/changes are supported.

The resource takes cyclic references into account.

Covered by existing tests (dozens of progressions, see TestExpectations).

* LayoutTests/platform/mac-sonoma-wk2-lbse-text/TestExpectations:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/W3C-SVG-1.1/masking-intro-01-f-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/W3C-SVG-1.1/masking-mask-01-b-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/absolute-root-position-masking-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/empty-mask-expected.txt: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/js-late-mask-and-object-creation-expected.txt: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/js-late-mask-creation-expected.txt: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/mask-changes-expected.txt: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/mask-child-changes-expected.txt: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/mask-colorspace-expected.txt: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/mask-excessive-malloc-expected.txt: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/mask-inside-defs-expected.txt: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/mask-invalidation-expected.txt: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/mask-with-all-units-expected.txt: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/mask-with-default-value-expected.txt: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/recursive-mask-expected.txt: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/transformedMaskFails-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/custom/visibility-override-mask-expected.txt: Added.
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/filters/filter-clip-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/foreignObject/mask-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/repaint/mask-clip-target-transform-expected.txt:
* LayoutTests/platform/mac-sonoma-wk2-lbse-text/svg/repaint/text-mask-update-expected.txt:
* LayoutTests/svg/repaint/mask-object-bounding-box-expected.txt: Added.
* LayoutTests/svg/repaint/mask-object-bounding-box-shrink-expected.txt: Added.
* LayoutTests/svg/repaint/mask-object-bounding-box-shrink.html: Added.
* LayoutTests/svg/repaint/mask-object-bounding-box-transformed-expected.txt: Added.
* LayoutTests/svg/repaint/mask-object-bounding-box-transformed.html: Added.
* LayoutTests/svg/repaint/mask-object-bounding-box.html: Added.
* LayoutTests/svg/repaint/mask-user-space-on-use-expected.txt: Added.
* LayoutTests/svg/repaint/mask-user-space-on-use-shrink-expected.txt: Added.
* LayoutTests/svg/repaint/mask-user-space-on-use-shrink.html: Added.
* LayoutTests/svg/repaint/mask-user-space-on-use-transformed-expected.txt: Added.
* LayoutTests/svg/repaint/mask-user-space-on-use-transformed.html: Added.
* LayoutTests/svg/repaint/mask-user-space-on-use.html: Added.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/rendering/RenderLayerModelObject.cpp:
(WebCore::RenderLayerModelObject::svgMaskerResourceFromStyle const):
(WebCore::RenderLayerModelObject::paintSVGClippingMask const):
(WebCore::RenderLayerModelObject::paintSVGMask const):
* Source/WebCore/rendering/RenderLayerModelObject.h:
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::isRenderSVGResourceContainer const):
(WebCore::RenderObject::isRenderSVGResourceMasker const):
* Source/WebCore/rendering/svg/RenderSVGContainer.cpp:
(WebCore::RenderSVGContainer::paint):
* Source/WebCore/rendering/svg/RenderSVGForeignObject.cpp:
(WebCore::RenderSVGForeignObject::paint):
* Source/WebCore/rendering/svg/RenderSVGImage.cpp:
(WebCore::RenderSVGImage::paint):
* Source/WebCore/rendering/svg/RenderSVGResourceMasker.cpp: Added.
(WebCore::RenderSVGResourceMasker::RenderSVGResourceMasker):
(WebCore::createImageBuffer):
(WebCore::RenderSVGResourceMasker::applyMask):
(WebCore::RenderSVGResourceMasker::resourceBoundingBox):
* Source/WebCore/rendering/svg/RenderSVGResourceMasker.h: Added.
* Source/WebCore/rendering/svg/RenderSVGResourceMaskerInlines.h: Added.
(WebCore::RenderSVGResourceMasker::maskElement const):
(WebCore::RenderSVGResourceMasker::maskUnits const):
(WebCore::RenderSVGResourceMasker::maskContentUnits const):
* Source/WebCore/rendering/svg/RenderSVGRoot.cpp:
(WebCore::RenderSVGRoot::paintObject):
* Source/WebCore/rendering/svg/RenderSVGShape.cpp:
(WebCore::RenderSVGShape::paint):
* Source/WebCore/rendering/svg/RenderSVGText.cpp:
(WebCore::RenderSVGText::paint):
* Source/WebCore/rendering/svg/SVGBoundingBoxComputation.cpp:
(WebCore::SVGBoundingBoxComputation::adjustBoxForClippingAndEffects const):
* Source/WebCore/rendering/svg/SVGRenderSupport.cpp:
(WebCore::SVGRenderSupport::paintSVGClippingMask): Deleted.
* Source/WebCore/rendering/svg/SVGRenderSupport.h:
* Source/WebCore/rendering/svg/SVGResources.cpp:
(WebCore::SVGResources::buildCachedResources):
* Source/WebCore/svg/SVGElement.cpp:
(WebCore::isSVGLayerAwareElement):
* Source/WebCore/svg/SVGMaskElement.cpp:
(WebCore::SVGMaskElement::createElementRenderer):
(WebCore::SVGMaskElement::calculateMaskContentRepaintRect):
* Source/WebCore/svg/SVGMaskElement.h:

Canonical link: https://commits.webkit.org/271153@main

mxschmitt committed:
[WinCairo] Read GITHUB_TOKEN from env when updating WebKitRequirements
https://bugs.webkit.org/show_bug.cgi?id=265373

Reviewed by Don Olmstead.

Read the API token from env and add it accordingly if set.

* Tools/Scripts/update-webkit-wincairo-libs.py:

Canonical link: https://commits.webkit.org/271154@main

eric-carlson committed:
[MediaStream] Address review comments after 270621@main
https://bugs.webkit.org/show_bug.cgi?id=265378
rdar://118832787

Reviewed by Youenn Fablet.

* Source/WebCore/Modules/mediastream/MediaStreamTrack.cpp:
(WebCore::MediaStreamTrack::applyConstraints): Resolve or reject promise in a task.
(WebCore::MediaStreamTrack::trackEnded): Wrap long comment.

* Source/WebCore/platform/mediastream/mac/AVVideoCaptureSource.mm:
(WebCore::AVVideoCaptureSource::resolvePendingPhotoRequest): Log an error message if there
is no producer.
(WebCore::AVVideoCaptureSource::rejectPendingPhotoRequest): Ditto.
(WebCore::AVVideoCaptureSource::captureOutputDidFinishProcessingPhoto): Return early if
capture fails.

* Source/WebCore/platform/mock/MockRealtimeVideoSource.h:
(WebCore::MockRealtimeVideoSource::DrawingState::DrawingState): Add `explicit`

* Source/WebKit/UIProcess/Cocoa/UserMediaCaptureManagerProxy.cpp:
(WebKit::UserMediaCaptureManagerProxy::takePhoto):

Canonical link: https://commits.webkit.org/271155@main

alanbaradlay committed:
[IFC][Integration] Move box geometry update to a dedicate class
https://bugs.webkit.org/show_bug.cgi?id=265387

Reviewed by Antti Koivisto.

This is in preparation for supporting intrinsic width computation on inline level boxes.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/layout/integration/LayoutIntegrationBoxTree.h:
* Source/WebCore/layout/integration/inline/LayoutIntegrationBoxGeometryUpdater.cpp: Added.
(WebCore::LayoutIntegration::BoxGeometryUpdater::BoxGeometryUpdater):
(WebCore::LayoutIntegration::BoxGeometryUpdater::updateReplacedDimensions):
(WebCore::LayoutIntegration::BoxGeometryUpdater::updateInlineBlockDimensions):
(WebCore::LayoutIntegration::BoxGeometryUpdater::updateInlineTableDimensions):
(WebCore::LayoutIntegration::BoxGeometryUpdater::updateListItemDimensions):
(WebCore::LayoutIntegration::BoxGeometryUpdater::updateListMarkerDimensions):
(WebCore::LayoutIntegration::contentLogicalWidthForRenderer):
(WebCore::LayoutIntegration::contentLogicalHeightForRenderer):
(WebCore::LayoutIntegration::horizontalLogicalMargin):
(WebCore::LayoutIntegration::verticalLogicalMargin):
(WebCore::LayoutIntegration::logicalBorder):
(WebCore::LayoutIntegration::logicalPadding):
(WebCore::LayoutIntegration::scrollbarLogicalSize):
(WebCore::LayoutIntegration::BoxGeometryUpdater::updateLayoutBoxDimensions):
(WebCore::LayoutIntegration::BoxGeometryUpdater::updateLineBreakBoxDimensions):
(WebCore::LayoutIntegration::BoxGeometryUpdater::updateInlineBoxDimensions):
(WebCore::LayoutIntegration::BoxGeometryUpdater::setGeometriesForLayout):
(WebCore::LayoutIntegration::BoxGeometryUpdater::updateInlineContentConstraints):
(WebCore::LayoutIntegration::BoxGeometryUpdater::rootLayoutBox const):
(WebCore::LayoutIntegration::BoxGeometryUpdater::rootLayoutBox):
* Source/WebCore/layout/integration/inline/LayoutIntegrationBoxGeometryUpdater.h: Added.
(WebCore::LayoutIntegration::BoxGeometryUpdater::takeNestedListMarkerOffsets):
(WebCore::LayoutIntegration::BoxGeometryUpdater::boxTree):
(WebCore::LayoutIntegration::BoxGeometryUpdater::boxTree const):
(WebCore::LayoutIntegration::BoxGeometryUpdater::layoutState):
(WebCore::LayoutIntegration::BoxGeometryUpdater::layoutState const):
* Source/WebCore/layout/integration/inline/LayoutIntegrationLineLayout.cpp:
(WebCore::LayoutIntegration::LineLayout::LineLayout):
(WebCore::LayoutIntegration::LineLayout::updateInlineContentDimensions):
(WebCore::LayoutIntegration::LineLayout::layout):
(WebCore::LayoutIntegration::LineLayout::updateInlineContentConstraints):
(WebCore::LayoutIntegration::LineLayout::updateReplacedDimensions): Deleted.
(WebCore::LayoutIntegration::LineLayout::updateInlineBlockDimensions): Deleted.
(WebCore::LayoutIntegration::LineLayout::updateInlineTableDimensions): Deleted.
(WebCore::LayoutIntegration::LineLayout::updateListItemDimensions): Deleted.
(WebCore::LayoutIntegration::LineLayout::updateListMarkerDimensions): Deleted.
(WebCore::LayoutIntegration::contentLogicalWidthForRenderer): Deleted.
(WebCore::LayoutIntegration::contentLogicalHeightForRenderer): Deleted.
(WebCore::LayoutIntegration::horizontalLogicalMargin): Deleted.
(WebCore::LayoutIntegration::verticalLogicalMargin): Deleted.
(WebCore::LayoutIntegration::logicalBorder): Deleted.
(WebCore::LayoutIntegration::logicalPadding): Deleted.
(WebCore::LayoutIntegration::scrollbarLogicalSize): Deleted.
(WebCore::LayoutIntegration::LineLayout::updateLayoutBoxDimensions): Deleted.
(WebCore::LayoutIntegration::LineLayout::updateLineBreakBoxDimensions): Deleted.
(WebCore::LayoutIntegration::LineLayout::updateInlineBoxDimensions): Deleted.
(WebCore::LayoutIntegration::LineLayout::prepareLayoutState): Deleted.
* Source/WebCore/layout/integration/inline/LayoutIntegrationLineLayout.h:

Canonical link: https://commits.webkit.org/271156@main

whsieh committed:
[UIAsyncTextInput] fast/forms/caps-lock-indicator-width.html fails with --use-async-uikit-interactions
https://bugs.webkit.org/show_bug.cgi?id=265357
rdar://118809857

Reviewed by Megan Gardner.

Fix this failing layout test when async UIKit interactions are enabled; see below.

* Source/WebCore/platform/ios/WebEvent.mm:
(webEventModifierFlags):

Add missing logic to handle `UIKeyModifierAlphaShift` (i.e. caps lock state) by adding the
`WebEventFlagMaskLeftCapsLockKey` modifier flag.

* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView insertionPointColor]):
(-[WKContentView selectionBarColor]):
(-[WKContentView selectionHighlightColor]):
(-[WKContentView textContentTypeForTesting]):

Replace internal calls into the private `-textInputTraits` method here, to avoid hitting our own
release assert.

Canonical link: https://commits.webkit.org/271157@main
